### PR TITLE
KACPER-11 added middleware with request validation

### PR DESCRIPTION
**Pull request by kmkcc2**

### CODE REVIEW

File: routes/users.js [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
 const { create, findAll, findOne, update, destroy } = require("../controllers/user.controller.js");
+const { findOneValidation, createValidation, putValidation, destroyValidation } = require("../middleware/users-requests-validations.js");
+const { validateRequest } = require("../middleware/request-validation.js");
+
 router
   .route("/:id")
-  .get(findOne)
-  .put(update)
-  .delete(destroy);
+  .get(...findOneValidation, validateRequest, findOne)
```
Comment (FilipAlbrycht) [inline routes/users.js]: ... is mandatory there?

File: middleware/users-requests-validations.js
Comment (FilipAlbrycht) [inline middleware/users-requests-validations.js:1]: i suggest to move validators to separeted directory

File: controllers/user.controller.js [hunk 2]
```diff
@@ -53,21 +42,14 @@ exports.findAll = async (req, res) => {
...
-    }
     const user = await User.findByPk(id);
     if (user) return res.send(user);
-
     return res.status(404).send({
       message: `Cannot find user with id: ${id}`,
     });
-  } catch (err) {
-    console.log(err.message);
+  }catch(err){
```
Comment (FilipAlbrycht) [inline controllers/user.controller.js]: always add whitespace before and after {}

File: middleware/request-validation.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const express = require('express');
+const { validationResult } = require('express-validator');
+
+exports.validateRequest = (req, res, next) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()){
+        return res.status(400).json({ errors: errors.array() })
+    }
+    console.log(errors.array())
```
Comment (FilipAlbrycht) [inline middleware/request-validation.js]: console.log not needed

**Review by FilipAlbrycht — APPROVED** [review]

**Review by benmotyka — CHANGES_REQUESTED** [review]
generalnie zmienilbym nieco sposob tego middleware tak zeby nie trzeba bylo zawsze wstawiac ...findOneValidation, validateRequest tylko jedna funkcje z parametrem, np cos w stylu: validateRequest(findOneValidation) albo np validateRequest([findOneValidation, anotherValidation])

File: routes/users.js (hunk 1 above)
Comment (kmkcc2) [reply routes/users.js]: I am checking if the provided id is an valid id, should I delete it?

**Review by benmotyka — APPROVED** [review]

**Pull request merged**